packages/ui/src/components/SymbiosisBarChart/index.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+
+import { Chart } from "../Charts";
+import type { ChartConfig, ChartDataPoint } from "../Charts/types";
+import type { SymbiosisBarChartProps } from "./types";
+import { cn } from "../../utils/cn";
+
+const SymbiosisBarChart = ({
+  data,
+  config,
+  xAxisFormatter,
+  tooltipLabelFormatter,
+  className,
+  legendClassName,
+  tooltipClassName,
+}: SymbiosisBarChartProps) => {
+
+  const defaultTickFormatter = React.useCallback((value: string) => {
+    const date = new Date(value);
+    return date.toLocaleDateString("en-US", {
+      month: "short",
+      day: "numeric",
+    });
+  }, []);
+
+  const defaultTooltipLabelFormatter = React.useCallback((value: string) => {
+    const date = new Date(value);
+    return date.toLocaleDateString("en-US", {
+      month: "short",
+      day: "numeric",
+      year: "numeric",
+    });
+  }, []);
+
```

Comment:
Do you believe that using useMemo could be efficient for the generateBars?

```
  const bars = useMemo(() => generateBars(data, config), [data, config]);
```

and then render the `bars`

Reply:
Hmm, the function is already extracted out of the component.
This means that the function `generateBars` is not re-created on every render.

The props that `generateBars` accepts are component props so the component will rerender if they change and thus also rerun the function.

Is the `useMemo` really necessary? I could be missing something 🤔

Reply:
You are right of course! Missed it, thought it was inside :)

Reply:
To be fair, the function is not re-created, but the array it returns actually gets re-created on every render. And as it returns an array of components the whole tree gets thrown away and recreated every time one of the props that the `SymbiosisBarChart` accepts changes (referentially). This also means that the `key` on the `Chart.Bar` essentially doesn't do anything. 

This is why it's not common practice to have functions that return components and not components that return components. React cannot automatically memoize them.

Reply:
Ok then. Let's create a component that returns components 👍😊

Reply:
Unfortunately Recharts uses things like `React.Children.map()` so the immediate children of the charts have to be specific Recharts components (like <Bar>, <Line>, etc.) for the them to be rendered, so I cannot extract the logic into another component.

I just do the loops and mapping in the father component.


[937e24a](https://github.com/synopsisapp/symbiosis-ui/pull/15/commits/937e24a19dcd40eadc5ba9bde1469fd34c532a07)